x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1058,10 +1057,10 @@ public ApiKeyDoc(
             long creationTime,
             long expirationTime,
             Boolean invalidated,
-            @Nullable String hash,
-            String name,
+            String hash,
+            @Nullable String name,
             int version,
-            @Nullable BytesReference roleDescriptorsBytes,
+            BytesReference roleDescriptorsBytes,
```

Comment:
Removed `Nullable` and replaced `optionalConstructorArg()` with `constructorArg()` for `hash` and `roleDescriptorsBytes` fields for clarity:
* The parser does not allow them to be null without something like `builder.declareStringOrNull(...)`. Therefore the `Nullable` etc does not have any effects anyway. In another word, the overall code logic is not impacted.
* Based on the logic in `ApiKeyService#newDocument`, these two fields can never be null.